tap_zendesk/client.py

```diff
@@ -129,8 +121,135 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         Yields:
             Each record from the source.
         """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
+        if not response.content:
+            self.logger.warning("Received empty response for URL: %s", response.url)
+            return
+
+        try:
+            response_json = response.json()
+            self.logger.debug(f"Response JSON: {response_json}")
+        except json.JSONDecodeError:
+            self.logger.error("Unable to decode JSON response: %s", response.text)
+            return
+
+        yield from extract_jsonpath(self.records_jsonpath, input=response_json)
+
+    def _request(
+        self,
+        prepared_request: PreparedRequest,
+        context: Optional[Dict[str, Any]] = None,
+    ) -> Response:
+        """Send a HTTP request with rate limiting.
+
+        Args:
+            prepared_request: The prepared HTTP request.
+            context: Stream partition or context dictionary.
+
+        Returns:
+            The HTTP response object.
+        """
+        # Calculate the time difference between the current time and the last request time
+        time_since_last_request = time.time() - self.last_request_time
+
+        # Ensure we don't exceed 250 requests per minute (1 request per 0.24 seconds)
+        if time_since_last_request < 0.24:
+            time.sleep(0.24 - time_since_last_request)
+
+        response = self.requests_session.send(
+            prepared_request,
+            timeout=self.timeout,
+            allow_redirects=self.allow_redirects,
+        )
+
+        self.last_request_time = time.time()
+
+        self._write_request_duration_log(
+            endpoint=self.path,
+            response=response,
+            context=context,
+            extra_tags={"url": prepared_request.path_url}
+            if self._LOG_REQUEST_METRIC_URLS
+            else None,
+        )
+
+        # Check for 404 Not Found and log a warning
+        if response.status_code == 404:
+            self.logger.warning(f"Received 404 for URL: {response.url}")
+            return None
+
+        self.validate_response(response)
+        self.logger.debug("Response received successfully.")
+        return response
+
+    def get_records(self, context):
```

Comment:
I think we don't need to override the get_records, we can just change the [get_new_paginator](https://github.com/meltano/sdk/blob/db38fcdd9f88dfca67870f07122e9333c2ccfac7/singer_sdk/streams/rest.py#L386) and `parse_response` to control the records to be to filter the end_date. All these request and parse should be done automatically by the sdk